src/main/java/gyro/aws/ec2/TransitGatewayMulticastDomainResource.java

```diff
@@ -0,0 +1,271 @@
+ *         transit-gateway: $(aws::transit-gateway example-transit-gateway)
+ *
+ *         association
+ *             vpc-attachment: $(aws::transit-gateway-vpc-attachment transit-gateway-vpc-attachment-example)
+ *             subnet: $(aws::subnet example-subnet-1)
+ *         end
+ *
+ *         group-member
+ *             group-ip-address: "224.0.0.0"
+ *             network-interface: $(aws::network-interface example-network-interface)
+ *         end
+ *     end
+ */
+@Type("transit-gateway-multicast-domain")
+public class TransitGatewayMulticastDomainResource extends Ec2TaggableResource<TransitGatewayMulticastDomain>
+    implements Copyable<TransitGatewayMulticastDomain> {
+
+    private TransitGatewayResource transitGateway;
+    private List<TransitGatewayMulticastDomainAssociationResource> association;
+    private List<TransitGatewayMulticastDomainGroupMemberResource> groupMember;
+    private List<TransitGatewayMulticastDomainGroupSourceResource> groupSource;
+
+    // Output
+
+    private String id;
+
+    /**
+     * The transit gateway that processes the multicast traffic. (Required)
+     */
+    @Required
+    public TransitGatewayResource getTransitGateway() {
+        return transitGateway;
+    }
+
+    public void setTransitGateway(TransitGatewayResource transitGateway) {
+        this.transitGateway = transitGateway;
+    }
+
+    /**
+     * The attachments and subnets to associate with the multicast domain.
+     *
+     * @subresource gyro.aws.ec2.TransitGatewayMulticastDomainAssociationResource
+     */
+    public List<TransitGatewayMulticastDomainAssociationResource> getAssociation() {
+        if (association == null) {
+            association = new ArrayList<TransitGatewayMulticastDomainAssociationResource>();
+        }
+        return association;
+    }
+
+    public void setAssociation(List<TransitGatewayMulticastDomainAssociationResource> association) {
+        this.association = association;
+    }
+
+    /**
+     * The network interfaces that receive the multicast traffic.
+     *
+     * @subresource gyro.aws.ec2.TransitGatewayMulticastDomainGroupMemberResource
+     */
+    public List<TransitGatewayMulticastDomainGroupMemberResource> getGroupMember() {
+        if (groupMember == null) {
+            groupMember = new ArrayList<TransitGatewayMulticastDomainGroupMemberResource>();
+        }
+        return groupMember;
+    }
+
+    public void setGroupMember(List<TransitGatewayMulticastDomainGroupMemberResource> groupMember) {
+        this.groupMember = groupMember;
+    }
+
+    /**
+     * The network interfaces that send the multicast traffic.
+     *
+     * @subresource gyro.aws.ec2.TransitGatewayMulticastDomainGroupSourceResource
+     */
+    public List<TransitGatewayMulticastDomainGroupSourceResource> getGroupSource() {
+        if (groupSource == null) {
+            groupSource = new ArrayList<TransitGatewayMulticastDomainGroupSourceResource>();
```

Comment:
Redundant to initialize an array list with type

```suggestion
            groupSource = new ArrayList<>();
```